common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -381,17 +381,19 @@ static void setMemory(Object o, long offset, long bytes, byte value) {
     }
 
     static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2, int length) {
-        if (length == 0) {
-            return true;
-        }
         final long baseOffset1 = BYTE_ARRAY_BASE_OFFSET + startPos1;
         final long baseOffset2 = BYTE_ARRAY_BASE_OFFSET + startPos2;
         final int remainingBytes = length & 7;
-        final long end = baseOffset1 + remainingBytes;
-        for (long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length; i >= end; i -= 8, j -= 8) {
-            if (UNSAFE.getLong(bytes1, i) != UNSAFE.getLong(bytes2, j)) {
-                return false;
-            }
+        if (length > 7) {
+            final long end = baseOffset1 + remainingBytes;
+            long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length;
+            do {
+                if (UNSAFE.getLong(bytes1, i) != UNSAFE.getLong(bytes2, j)) {
```

Comment:
>Sorry to be nitpicky, but my intuition says that this will be slower than the forward path just because it would be easier for the cache to take advantage of this.

> Is there benchmarking for this method?

`PlatformDependentBenchmark` exists which I will run for a before/after. This was shown to provide benefits on smaller inputs when running benchmarks on `equalsConstantTime`. The overhead of allocating `i`, `j`, and `end` can be avoided for smaller inputs that don't need these variables.

> Also, how are you sure that byte arrays are word aligned?

`PlatformDependent.equalsConstantTime` checks if aligned access is allowed in general, but no check is done on the individual arrays. Do you know how I would check, and an example allocation which I can use to verify this method?